database.c

```diff
@@ -210,7 +211,27 @@ static void print_replay_statistics(int dry_run) {
 	busy_time = stat_exec.tv_usec / 1000000.0 + stat_exec.tv_sec;
 	/* calculate total session time */
 	session_time = stat_session.tv_usec / 1000000.0 + stat_session.tv_sec;
-
+	/*Calculate lengh of the replay file*/
+	timersub(&last_stmt_time, &first_stmt_time, &total_length);
+	length_days = total_length.tv_sec / 86400;
+	total_length.tv_sec -= length_days * 86400;
+	length_hours = total_length.tv_sec / 3600;
+	total_length.tv_sec -= length_hours * 3600;
+	length_minutes = total_length.tv_sec / 60;
+	total_length.tv_sec -= length_minutes * 60;
+	length_seconds = total_length.tv_sec + total_length.tv_usec / 1000000.0;
+
+	fprintf(sf, "Replay length:");
```

Comment:
"Replay length" is confusing.

Wouldn't "Original run time" or "Duration of recorded workload" describe it better?

The information is not really useful if you replay against a database:

- If replay is delayed, the maximum delay will already give you a good idea how fast the original run might have been.
- If replay is faster than the original, the difference will be about zero &mdash; remember that pgreplay does not replay a statement earlier than the original time when it ran.

So I think that this should only be displayed during a dry run.

On the other hand, it would be useful for `file_consumer`.
Any chance you could add that to `print_parse_statistics` as well?

Reply:
I'll look into that also :)